### PR TITLE
Dashboard with almost all of the graphs

### PR DESCRIPTION
**Pull request by paulauus**
I also added the colour changes that the coaches suggested. Going to make a lot of changes, including separating graph functions to their own script. Next I'm going to create that Games page - already added a hyperlink to it here.

### CODE REVIEW

File: dashboard/.streamlit/config.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [theme]
 base = "dark"
 primaryColor = "#6a0dad"  # Purple primary color
-backgroundColor = "#121212" # Dark background color
-secondaryBackgroundColor = "#1e1e1e" # Slightly lighter background
-textColor = "#ffffff"       # White text color
+backgroundColor = "#1e1e1e" # Dark background color
+secondaryBackgroundColor = "#121212" # Slightly darker background
```
Comment (Leon-JS02) [inline dashboard/.streamlit/config.toml:5]: These are nice colours

File: dashboard/dashboard.py [hunk 2]
```diff
@@ -7,135 +8,99 @@
...
-import pandas as pd
-import altair as alt
+import nltk
 
-from sl_queries import get_game_data, get_weekdays_data, get_daily_game_count
+from functions import create_donut_chart, create_genre_bar_chart, create_line_chart, create_os_bar_chart, create_platform_bar_chart, create_release_line_chart, display_game_table, create_word_cloud, preprocess_descriptions
+from sl_queries import get_game_descriptions
+
+# Download the 'punkt' tokenizer model
+nltk.download('punkt_tab')
```
Comment (Leon-JS02) [inline dashboard/dashboard.py:17]: Fancy integration

File: dashboard/functions.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+import re
+from collections import Counter
+import numpy as np
+
+import streamlit as st
+import pandas as pd
+import altair as alt
+from nltk.corpus import stopwords as nltk_stopwords
+from nltk.tokenize import word_tokenize
```
Comment (mehrdadhalali) [inline dashboard/functions.py:11]: Nice tokenising

File: dashboard/dashboard.py [hunk 4]
```diff
@@ -7,135 +8,99 @@
...
-    columns={
-        'game_name': 'Title',
-        'game_genres': 'Genres',
-        'release_date': 'Release Date',
-        'platform': 'Platform',
-        'release_price': 'Price',
-        'listing_url': 'Listing URL'  # Ensure listing_url is included for hyperlinks
+    """
+    <style>
+    a {
```
Comment (Leon-JS02) [inline dashboard/dashboard.py:25]: !important?

File: dashboard/dashboard.py [hunk 5]
```diff
@@ -7,135 +8,99 @@
...
+    st.subheader("Top Keywords in Game Descriptions")
+    st.markdown(
+        "<style>.line { border: 0.5px solid; margin: 0; }</style><div class='line'></div>",
+        unsafe_allow_html=True)
+    
+    descriptions = get_game_descriptions(
+            show_nsfw, start_date, end_date, os_selection)
+    word_counts = preprocess_descriptions(descriptions)
+    word_cloud_fig = create_word_cloud(word_counts)
+    st.pyplot(word_cloud_fig, use_container_width=True)
```
Comment (Leon-JS02) [inline dashboard/dashboard.py:105]: Nice word cloud

File: dashboard/functions.py [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+            text='Daily Count of Unique Game Releases',
+            anchor='middle',
+            fontSize=17
+        )
+    )
+
+    return line_chart
+
+
+def display_game_table(show_nsfw, os_selection, start_date, end_date, search_query):
```
Comment (Leon-JS02) [inline dashboard/functions.py:75]: Nice compartmentalised code

File: dashboard/sl_queries.py
Comment (mehrdadhalali) [inline dashboard/sl_queries.py:1]: Perhaps in the future we could make one call to the database, and edit the dataframe instead.

**Review by mehrdadhalali — APPROVED** [review]

File: dashboard/sl_queries.py [hunk 7]
```diff
@@ -35,34 +35,92 @@ def get_game_data(show_nsfw):
...
         gl.release_price,
-        gl.listing_url
+        gl.listing_url,
+        STRING_AGG(DISTINCT os.os_name, ', ') AS os_name
     FROM game g
     LEFT JOIN game_genre_assignment ga ON g.game_id = ga.game_id
     LEFT JOIN genre ge ON ga.genre_id = ge.genre_id
     LEFT JOIN game_listing gl ON g.game_id = gl.game_id
     LEFT JOIN platform p ON gl.platform_id = p.platform_id
+    LEFT JOIN game_os_assignment goa ON g.game_id = goa.game_id
```
Comment (Leon-JS02) [inline dashboard/sl_queries.py:45]: Monster query

**Review by Leon-JS02 — APPROVED** [review]
Great dashboarding work. The word cloud is a fancy feature.

**Pull request merged**